src/runtime/runtime.js

```diff
@@ -84,43 +87,46 @@ export function getRuntime() {
 }
 
 /**
+ * Installs SwG runtime.
  * @param {!Window} win
  */
 export function installRuntime(win) {
+  // Only install the SwG runtime once.
   if (win[RUNTIME_PROP] && !isArray(win[RUNTIME_PROP])) {
     return;
   }
 
+  // Create a SwG runtime.
   const runtime = new Runtime(win);
 
-  const waitingArray = [].concat(win[RUNTIME_PROP], win[RUNTIME_LEGACY_PROP]);
-
-  // Public runtime.
+  // Create a public version of the SwG runtime.
   const publicRuntime = createPublicRuntime(runtime);
 
-  const dependencyInstaller = {};
-
   /**
+   * Executes a callback when SwG runtime is ready.
    * @param {function(!Subscriptions)} callback
    */
-  function pushDependency(callback) {
+  function push(callback) {
     if (!callback) {
       return;
     }
+
     runtime.whenReady().then(() => {
       callback(publicRuntime);
     });
   }
-  Object.defineProperty(dependencyInstaller, 'push', {
-    get: () => pushDependency,
-    configurable: false,
-  });
-  win[RUNTIME_PROP] = dependencyInstaller;
-  win[RUNTIME_LEGACY_PROP] = dependencyInstaller;
-  if (waitingArray) {
-    waitingArray.forEach(pushDependency);
-  }
+
+  // Queue up any callbacks the publication might have provided.
+  [].concat(win[RUNTIME_PROP], win[RUNTIME_LEGACY_PROP]).forEach(push);
```

Comment:
Nit: It would be marginally more readable if you left the `const waitingArray =`

Reply:
How about the var name `waitingCallbacks`?

Reply:
Would the var name `const waitingCallbacks` work well?

Reply:
sure